tokio-boring/src/lib.rs

```diff
@@ -307,6 +307,58 @@ impl<S> HandshakeError<S> {
             _ => None,
         }
     }
+
+    /// Returns a reference to the inner SSL error, if this error was caused by
+    /// a SSL error.
+    pub fn as_ssl_error(&self) -> Option<&boring::error::ErrorStack> {
+        match &self.0 {
+            ssl::HandshakeError::Failure(s) => s.error().ssl_error(),
+            ssl::HandshakeError::SetupFailure(ref s) => Some(s),
+            _ => None,
```

Comment:
Shouldn't this also handle `HandshakeError::WouldBlock`? Or is it impossible somehow for WouldBlock to be an SSL error?
```suggestion
            ssl::HandshakeError::Failure(s) | ssl::HandshakeError::WouldBlock(s) => s.error().ssl_error(),
            ssl::HandshakeError::SetupFailure(ref s) => Some(s),
```